spec/services/idr_profile_querier_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+REQUIRED_CONFIG_PARAMS =
+  %w(PROFILE_REQUEST_URL PROFILE_REQUEST_QUERY_PARAM PROFILE_REQUEST_FIRST_NAME
+     PROFILE_REQUEST_LAST_NAME PROFILE_REQUEST_EMAIL PROFILE_REQUEST_GENDER
+     PROFILE_REQUEST_CLASS_YEAR PROFILE_REQUEST_COLLEGE).freeze
+DUMMY_XML_RESPONSE =
+  <<~HEREDOC
+    <?xml version="1.0" encoding="UTF-8"?>
+    <ServiceResponse>
+      <Record>
+          <Login>foo</Login>
+          <Gender>F</Gender>
+          <FirstName>Jane</FirstName>
+          <LastName>Smith</LastName>
+          <Email>jane.smith@example.com</Email>
+          <AltId>123456</AltId>
+          <ClassYear>2018</ClassYear>
+          <College>AB</College>
+      </Record>
+    </ServiceResponse>
+  HEREDOC
+
+RSpec.describe IdrProfileQuerier do
+  describe '.query' do
+    it 'calls :query on an instance of IdrProfileQuerier' do
+      profile_querier = mock_profile_querier(id: 'foo')
+      described_class.query(id: 'foo')
+      expect(profile_querier).to have_received(:query)
+    end
+  end
+
+  describe '#query' do
+    context 'success' do
+      before do
+        assign_required_attributes
+        stub_profile_request
+      end
+      it 'returns profile data with properly mapped gender' do
+        result = described_class.query(id: 'bar')
+        expected = { first_name: 'Jane', last_name: 'Smith',
+                     email: 'jane.smith@example.com', gender: 'female',
+                     class_year: '2018', college: 'AB' }
```

Comment:
this hash should maybe also be a fixture? or at least stored in a way that makes it obvious that that particular xml is where these attributes come from